test/scripts/e2e_subs/tealprogs/cross-round.teal

```diff
@@ -0,0 +1,183 @@
+#pragma version 2
+
+// This program writes some data on optin and then
+// validates stored values on subsequent invocations.
+// Invocation parameter is controlled by app arg ("first", "second", "third").
+// The program also establishes behavior for reading and deleting non-existing keys.
+
+// allow app creation and deletion
+txn ApplicationID
+int 0
+==
+bnz success
+
+txn OnCompletion
+int DeleteApplication
+==
+bnz success
+
+txn OnCompletion
+int OptIn
+==
+txna ApplicationArgs 0
+byte "first"
+==
+&&
+bnz first
+txna ApplicationArgs 0
+byte "second"
+==
+bnz second
+txna ApplicationArgs 0
+byte "third"
+==
+bnz third
+b fail
+
+// check non-existing keys deletion and reading
+// write some initial values
+first:
+// check global non-existing
+int 0
+byte "foo"
+app_global_get_ex
+// value may not exist
+int 1
+==
+bnz fail
+// keep one zero value (non-initialized value) on the stack from app_global_get_ex
+byte "foo"
+app_global_get
+// check local non-existing
+// re-use 0 in the top of the stack as account idx = 0 => sender
+// re-use 0 from the stack as app idx
+// this both are done by purpose to coin possible side effects of non-initialized stack values
+byte "bar"
+app_local_get_ex
+// value may not exist
+int 1
+==
+bnz fail
+// re-use 0 in the top of the stack as account idx = 0 => sender
+byte "bar"
+app_local_get
+// keep one zero value (non-initialized value) on the stack from app_local_get
+byte "foobar"
+app_global_del
+// re-use 0 in the top of the stack as account idx = 0 => sender
+byte "barfoo"
+app_local_del
+
```

Comment:
It may be worthwhile to check the output of `app_opted_in` here as well. I'm not 100% sure but I think it should return 1 during the optin call. Either way it would be good to pin this behavior down.

Reply:
yes, this is exact problem I'm fixing in **refactor** branch. It must return `1` during optin call. Will add it here.

Reply:
done